sensors.py

```diff
@@ -20,28 +25,93 @@ def get_reading(self, sensor_type: SensorType) -> int | float:
         raise NotImplementedError
 
 class DHT(Sensor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.dht = DHT22(board.D4)
+
     def get_reading(self, sensor_type: SensorType) -> int:
         with self._lock:
-            if sensor_type == SensorType.TEMPERATURE:
-                return randint(10, 30)
-            if sensor_type == SensorType.HUMIDITY:
-                return randint(40, 100)
+            try:
+                if sensor_type == SensorType.TEMPERATURE:
+                    return self.dht.temperature
+                if sensor_type == SensorType.HUMIDITY:
+                    return self.dht.humidity
+            except:
+                logging.exception("Couldn't get a reading")
+                raise SensorReadingError
         raise WrongSensorType
 
 class BMP280(Sensor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.bmp280 = adafruit_bmp280.Adafruit_BMP280_I2C(board.I2C(), address=0x76)
+        
     def get_reading(self, sensor_type: SensorType) -> int | float:
         with self._lock:
-            if sensor_type == SensorType.PRESSURE:
-                return randint(990, 1100)
+            try:
+                if sensor_type == SensorType.PRESSURE:
+                    return int(self.bmp280.pressure)
+            except:
```

Comment:
I understand that we want to catch both `AssertionError` and `ValueError`?
Can't we then
`except .../* as ecx`